src/WebJobs.Script/Config/ScriptJobHostOptionsSetup.cs

```diff
@@ -73,6 +74,16 @@ public void Configure(ScriptJobHostOptions options)
             options.TestDataPath = webHostOptions.TestDataPath;
             options.IsFileSystemReadOnly = webHostOptions.IsFileSystemReadOnly;
             options.IsStandbyConfiguration = webHostOptions.IsStandbyConfiguration;
+
+            var telemetryModeSection = jobHostSection.GetSection(ConfigurationSectionNames.TelemetryMode);
```

Comment:
This might work?
```suggestion
            options.TelemetryMode = jobHostSection.GetValue(ConfigurationSectionNames.TelemetryMode, TelemetryMode.ApplicationInsights);
```

Can then remove the `if`/`else` statement

Reply:
I will have to use try catch block with `GetValue` as it throws argument exception. Leaving this as-is for now.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.getvalue?view=dotnet-plat-ext-8.0#microsoft-extensions-configuration-configurationbinder-getvalue-1(microsoft-extensions-configuration-iconfiguration-system-string-0)

What argument exception will it throw?

Reply:
yes, `jobHostSection.GetValue(ConfigurationSectionNames.TelemetryMode, TelemetryMode.ApplicationInsights);` is throwing an exception if string is not a valid Enum.